sdk/keyvault/azure_security_keyvault_keys/README.md

```diff
@@ -1,66 +1,370 @@
 
 ## Examples
 
-### Creating a key
+The following section provides several code snippets using the `KeyClient`, covering some of the most common Azure Key Vault keys service related tasks:
+
+* [Create a key](#create-a-key)
+* [Retrieve a key](#retrieve-a-key)
+* [Update an existing key](#update-an-existing-key)
+* [Delete a key](#delete-a-key)
+* [List keys](#list-keys)
+
+### Create a key
+
+`create_key` creates a Key Vault key to be stored in the Azure Key Vault. If a key with the same name already exists, then a new version of the key is created.
 
 ```rust no_run
 use azure_identity::DefaultAzureCredential;
-use azure_security_keyvault_keys::{models::KeyCreateParameters, KeyClient};
+use azure_security_keyvault_keys::{
+    models::{KeyBundle, KeyCreateParameters, JsonWebKeyCurveName, JsonWebKeyType},
+    ResourceExt, KeyClient,
+};
 
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let credential = DefaultAzureCredential::new()?;
+    let client = KeyClient::new(
+        "https://your-key-vault-name.vault.azure.net/",
+        credential.clone(),
+        None,
+    )?;
+
+    // Create an EC key.
+    let body = KeyCreateParameters {
+        kty: Some(JsonWebKeyType::EC),
+        curve: Some(JsonWebKeyCurveName::P256),
+        ..Default::default()
+    };
+
+    let key = client
+        .create_key("key-name", body.try_into()?, None)
+        .await?
+        .into_body()
+        .await?;
 
+    println!(
+        "Key Name: {:?}, Type: {:?}, Version: {:?}",
+        key.resource_id()?.name,
+        key.key.as_ref().map(|k| k.kty.as_ref()),
+        key.resource_id()?.version,
+    );
+
+    Ok(())
+}
+```
+
+### Retrieve a key
+
+`get_key` retrieves a public key (or only metadata for symmetric keys) previously stored in the Azure Key Vault. Setting the `key-version` to an empty string will return the latest version.
+
+```rust no_run
+use azure_identity::DefaultAzureCredential;
+use azure_security_keyvault_keys::{models::KeyBundle, KeyClient};
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let credential = DefaultAzureCredential::new()?;
     let client = KeyClient::new(
-        "https://<your-key-vault-name>.vault.azure.net",
+        "https://your-key-vault-name.vault.azure.net/",
         credential.clone(),
         None,
     )?;
 
+    // Retrieve a public key as a JWK using the key client.
     let key = client
-        .create_key("my-key", KeyCreateParameters::default().try_into()?, None)
+        .get_key("key-name".into(), "key-version".into(), None)
+        .await?
+        .into_body()
+        .await?;
+
+    println!("Key: {:#?}", key.key);
+
+    Ok(())
+}
+```
+
+### Update an existing key
+
+`update_key` updates a key previously stored in the Azure Key Vault. Only the attributes of the key are updated. To update the value, call `KeyClient::create_key` on a key with the same name.
```

Comment:
Should this be [`KeyClient::update_key()`] to force a link?

Reply:
Not in a README. That only works in rustdocs. While included in lib.rs and that *might* happen, it wouldn't render as a proper link here in the README and would fail the link checker.